pycqed/measurement/waveform_control/segment.py

```diff
@@ -974,9 +987,28 @@ def calculate_hash(self, elname, codeword, channel):
 
         for pulse in self.elements[elname]:
             if pulse.codeword in {'no_codeword', codeword}:
-                hashlist += pulse.hashables(tstart, channel)
+                hashlist += self.hashables(pulse, tstart, channel)
         return tuple(hashlist)
 
+    @staticmethod
+    def hashables(pulse, tstart, channel):
```

Comment:
Without any code comment, I am not sure I understand why we had to do this change but I trust that it has been tested on the XLD setup.